[gherkin-c] Build error on x64-osx

Package: gherkin-c[core]:x64-osx -> 2019-10-07#2
**Host Environment**
- Host: arm64-osx
- Compiler: AppleClang 13.0.0.13000029
-    vcpkg-tool version: 2999-12-31-unknownhash-debug
    vcpkg-scripts version: 4fff1e9cf 2022-01-17 (10 minutes ago)


**To Reproduce**
`vcpkg install `


**Failure logs**
```
-- Using cached c-libs-gherkin-c-dd180edc7d092311f2e90a0c4957061156d39dd3.tar.gz.
-- Extracting source /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/c-libs-gherkin-c-dd180edc7d092311f2e90a0c4957061156d39dd3.tar.gz
-- Using source at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/src/1156d39dd3-34edf2e342.clean
-- Found external ninja('1.10.2').
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/tools/cmake-3.21.1-osx/cmake-3.21.1-macos-universal/CMake.app/Contents/bin/cmake /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/src/1156d39dd3-34edf2e342.clean -DBUILD_GHERKIN_TESTS=OFF -DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg -DZ_VCPKG_ROOT_DIR=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg -D_VCPKG_INSTALLED_DIR=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed -DVCPKG_MANIFEST_INSTALL=OFF -DCMAKE_OSX_ARCHITECTURES=x86_64 -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/gherkin-c_x64-osx/debug
    Working Directory: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/config-x64-osx-dbg-out.log
      /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:358 (vcpkg_execute_required_process)
  ports/gherkin-c/portfile.cmake:9 (vcpkg_configure_cmake)
  scripts/ports.cmake:145 (include)



```
<details><summary>/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/config-x64-osx-dbg-out.log</summary>

```
-- The C compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
-- Generating done
```
</details>
<details><summary>/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/config-x64-osx-dbg-err.log</summary>

```
CMake Error:
  Running

   '/opt/homebrew/bin/ninja' '-C' '/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/gherkin-c/x64-osx-dbg' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:588: bad $-escape (literal $ must be written as $$)

  



CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_LIBDIR
    VCPKG_CRT_LINKAGE
    VCPKG_PLATFORM_TOOLSET
    VCPKG_SET_CHARSET_FLAG
    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
```
</details>

**Additional context**
<details><summary>vcpkg.json</summary>

```
{
  "name": "lichtsteuerung",
  "version-string": "2.0.0",
  "dependencies": [
    "gherkin-c"
  ]
}

```
</details>

--- a/ports/gherkin-c/fix-include-path.patch
+++ b/ports/gherkin-c/fix-include-path.patch
@@ -7,7 +7,7 @@ index 74b9ffb..6af4645 100644
  
  add_library(gherkin ${GHERKIN_SRS})
 -target_include_directories(gherkin PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_SOURCE_DIR}/src>")
-+target_include_directories(gherkin PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_SOURCE_DIR}/src>" "$<INSTALL_INTERFACE:include/gherkin_c>")
++target_include_directories(gherkin PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_SOURCE_DIR}/src>" "$<INSTALL_INTERFACE:include/gherkin-c>")
  
  if (BUILD_GHERKIN_TESTS)
    LIST(APPEND GENERATE_TOKEN
@@ -16,7 +16,7 @@ index 74b9ffb..6af4645 100644
  endif()
  
 -set(include_install_dir "include")
-+set(include_install_dir "include/gherkin_c")
++set(include_install_dir "include/gherkin-c")
  set(lib_install_dir "lib/")
  set(bin_install_dir "bin/")
  set(config_install_dir "lib/cmake/")

--- a/ports/gherkin-c/fix-include-path.patch
+++ b/ports/gherkin-c/fix-include-path.patch
@@ -1,0 +1,22 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 74b9ffb..6af4645 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -60,7 +60,7 @@ LIST(APPEND GHERKIN_SRS
+         )
+ 
+ add_library(gherkin ${GHERKIN_SRS})
+-target_include_directories(gherkin PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_SOURCE_DIR}/src>")
++target_include_directories(gherkin PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_SOURCE_DIR}/src>" "$<INSTALL_INTERFACE:include/gherkin_c>")
+ 
+ if (BUILD_GHERKIN_TESTS)
+   LIST(APPEND GENERATE_TOKEN
+@@ -96,7 +96,7 @@ if (BUILD_GHERKIN_TESTS)
+   ENDFOREACH()
+ endif()
+ 
+-set(include_install_dir "include")
++set(include_install_dir "include/gherkin_c")
+ set(lib_install_dir "lib/")
+ set(bin_install_dir "bin/")
+ set(config_install_dir "lib/cmake/")

--- a/ports/gherkin-c/fix-install-error.patch
+++ b/ports/gherkin-c/fix-install-error.patch
@@ -1,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 3227bf5..74b9ffb 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -131,7 +131,3 @@ install(
+ #    NAMESPACE "${namespace}"
+ #    DESTINATION "${config_install_dir}")
+ 
+-add_custom_target(install_${PROJECT_NAME}
+-        $(MAKE) install
+-        DEPENDS gherkin
+-        COMMENT "Installing ${PROJECT_NAME}")

--- a/ports/gherkin-c/portfile.cmake
+++ b/ports/gherkin-c/portfile.cmake
@@ -4,19 +4,20 @@ vcpkg_from_github(
     REF dd180edc7d092311f2e90a0c4957061156d39dd3
     SHA512 c6b38ab0e7a0fd1061d86b0ff9d9140f8c3d6f15cfc1673e947254c6c03a66d3e6aae5b267b074aa10fa30ba2850190c9e9ea7c12e340e4f8c5575b9bf31bab3
     HEAD_REF master
+    PATCHES
+        fix-install-error.patch
+        fix-include-path.patch
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS 
         -DBUILD_GHERKIN_TESTS=OFF
 )
-vcpkg_install_cmake()
-
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-
+vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/gherkin-c RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/gherkin-c" RENAME copyright)

--- a/ports/gherkin-c/vcpkg.json
+++ b/ports/gherkin-c/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "gherkin-c",
-  "version-string": "2019-10-07",
+  "version-date": "2019-10-07",
   "port-version": 3,
   "description": "Gherkin parser/compiler in C",
   "dependencies": [

--- a/ports/gherkin-c/vcpkg.json
+++ b/ports/gherkin-c/vcpkg.json
@@ -1,6 +1,12 @@
 {
   "name": "gherkin-c",
   "version-string": "2019-10-07",
-  "port-version": 2,
-  "description": "Gherkin parser/compiler in C"
+  "port-version": 3,
+  "description": "Gherkin parser/compiler in C",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    }
+  ]
 }

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -363,15 +363,15 @@ gflags:arm-uwp=fail
 gflags:x64-uwp=fail
 
 # Conflicts with libevent
-gherkin-c:arm64-windows      = skip
-gherkin-c:arm-uwp            = skip
-gherkin-c:x64-linux=fail
-gherkin-c:x64-osx=fail
-gherkin-c:x64-uwp            = skip
-gherkin-c:x64-windows        = skip
-gherkin-c:x64-windows-static = skip
-gherkin-c:x64-windows-static-md=skip
-gherkin-c:x86-windows        = skip
+gherkin-c:arm64-windows         = skip
+gherkin-c:arm-uwp               = skip
+gherkin-c:x64-linux             = skip
+gherkin-c:x64-osx               = skip
+gherkin-c:x64-uwp               = skip
+gherkin-c:x64-windows           = skip
+gherkin-c:x64-windows-static    = skip
+gherkin-c:x64-windows-static-md = skip
+gherkin-c:x86-windows           = skip
 gl3w:arm64-windows=fail
 gl3w:arm-uwp=fail
 glew:arm64-windows=fail

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2442,7 +2442,7 @@
     },
     "gherkin-c": {
       "baseline": "2019-10-07",
-      "port-version": 2
+      "port-version": 3
     },
     "giflib": {
       "baseline": "5.2.1",

--- a/versions/g-/gherkin-c.json
+++ b/versions/g-/gherkin-c.json
@@ -1,8 +1,8 @@
 {
   "versions": [
     {
-      "git-tree": "27d09a0f7885f6f47ccc1a83b9103fb55422999e",
-      "version-string": "2019-10-07",
+      "git-tree": "87141daed8af9dab4f1891a0f5de8258ca866273",
+      "version-date": "2019-10-07",
       "port-version": 3
     },
     {

--- a/versions/g-/gherkin-c.json
+++ b/versions/g-/gherkin-c.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "87141daed8af9dab4f1891a0f5de8258ca866273",
+      "git-tree": "cb7cfa3a82651ab27840177189e5dc7938bf1241",
       "version-date": "2019-10-07",
       "port-version": 3
     },

--- a/versions/g-/gherkin-c.json
+++ b/versions/g-/gherkin-c.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "27d09a0f7885f6f47ccc1a83b9103fb55422999e",
+      "version-string": "2019-10-07",
+      "port-version": 3
+    },
+    {
       "git-tree": "e59a73d00ecbf68ae14b61beec141128d8f94612",
       "version-string": "2019-10-07",
       "port-version": 2

